### PR TITLE
Fix ColumnUnknownExceptions from object columns of system tables

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB
On a master node started with `-Cnode.attr.storage=hot`:
```
cr> select attributes from sys.nodes;
+--------------------+
| attributes         |
+--------------------+
| {"storage": "hot"} |
+--------------------+
SELECT 1 row in set (0.083 sec)

cr> select attributes['storage'] from sys.nodes;
ColumnUnknownException[Column attributes['storage'] unknown]
```
An exception is thrown eventhough attributes['storage'] exists. This is because sys.nodes table's attribute column's sub-columns info is not available locally(`attributes` is available while `attributes['storage']` is not) but need to be retreived via NodeStatsAction.

I think there are two ways to handle this 1) we cat treat objects from system tables as ignored (since the table schema is not fully available) or 2) do not throw and fall back to subscript function, `subscript(attribute, 'storage')` and execute it if the column is from system tables.

Fixes https://github.com/crate/crate/issues/17102 which is a regression caused by https://github.com/crate/crate/pull/17052.

## Checklist

 - [x] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by jeeminso** [comment]
There is a single test failure with 1). It is due to comparison against an object literal which is DYNAMIC by default causing the object from sys table to be cast from IGNORED > DYNAMIC.

**Comment by jeeminso** [comment]
There is also a single test failure from option 2). But it seems like a testcase problem.

**Comment by jeeminso** [comment]
Hi @seut, I think there are two ways to solve this, the first and the third commit. Could you take a look at them?

**Review by seut — COMMENTED** [review]
~Shouldn't we rather make `sys.nodes.attributes` a typed object?~
Ignore this please, it's a user-defined map 🤦

**Review by seut — COMMENTED** [review]
Looks good.
I spotted the usage of `UNTYPED` also at https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/metadata/sys/SysSegmentsTableInfo.java#L66, could you fix this as well please? Thanks!

File: server/src/main/java/io/crate/metadata/sys/SysNodesTableInfo.java [hunk 1]
```diff
@@ -102,7 +103,7 @@ public static class Columns {
         .add("name", STRING, NodeStatsContext::name)
         .add("hostname", STRING, NodeStatsContext::hostname)
         .add("rest_url", STRING, NodeStatsContext::restUrl)
-        .add("attributes", UNTYPED_OBJECT, NodeStatsContext::attributes)
+        .add("attributes", ObjectType.of(ColumnPolicy.IGNORED).build(), NodeStatsContext::attributes)
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/metadata/sys/SysNodesTableInfo.java:106]:
```suggestion
        .addDynamicObject("attributes", STRING, NodeStatsContext::attributes)
```

File: server/src/testFixtures/java/io/crate/testing/SQLResponseAssert.java [hunk 2]
```diff
@@ -95,4 +95,9 @@ public SQLResponseAssert isEmpty() {
         assertThat(actual.rows()).isEmpty();
         return this;
     }
+
+    public SQLResponseAssert isNotEmpty() {
+        assertThat(actual.rows()).isNotEmpty();
+        return this;
+    }
```
Comment (mfussenegger) [inline server/src/testFixtures/java/io/crate/testing/SQLResponseAssert.java:102]:
```suggestion
```

We should assert the actual expected result if possible

File: server/src/test/java/io/crate/integrationtests/SysNodesITest.java [hunk 3]
```diff
@@ -92,4 +92,10 @@ public void test_filter_on_not_selected_column_on_sys_nodes_returns_record() thr
         execute("select fs from sys.nodes where name = 'node_s0'");
         assertThat(response.rowCount()).isEqualTo(1L);
     }
+
+    @Test
+    public void test_can_select_node_attributes_sub_fields() {
+        execute("SELECT attributes['color'] FROM sys.nodes");
+        assertThat(response).isNotEmpty();
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/SysNodesITest.java:99]:
```suggestion
        execute("SELECT attributes['color'] FROM sys.nodes order by 1");
        assertThat(response).hasRows(
            "blue",
            "red"
        );
```

**Review by mfussenegger — CHANGES_REQUESTED** [review]

**Comment by jeeminso** [comment]
Thank you for reviewing @seut @mfussenegger . Please check for the updates.

**Review by mfussenegger — APPROVED** [review]

**Comment by jeeminso** [comment]
@mergifyio requeue

**Pull request merged**